src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +117,137 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+
+                    if (element.has("all")) {
+                        JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
+                    } else if (element.has("results")) {
+                        JSONArray entriesArray = element.getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
                     }
                 }
-            } catch (JSONException | TokenMgrException e) {
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry(StandardEntryType.Article);
+
+            // Set the author and keywords field
+            Optional<String> authors = toAuthors(item.optJSONArray("authors"));
+            authors.ifPresent(value -> entry.setField(StandardField.AUTHOR, value));
+
+            Optional<String> keywords = Optional.ofNullable(getKeywords(item.optJSONObject("primaryClass")));
+            keywords.ifPresent(value -> entry.setField(StandardField.KEYWORDS, value));
+
+            // Set the rest of the fields based on the mappings
+            for (Map.Entry<StandardField, List<String>> mapEntry : FIELD_MAPPINGS.entrySet()) {
+                StandardField field = mapEntry.getKey();
+                List<String> path = mapEntry.getValue();
+
+                // Skip author and keywords fields as they are already set
+                if (field == StandardField.AUTHOR || field == StandardField.KEYWORDS) {
+                    continue;
+                }
+
+                Optional<String> value = getOthers(item, path);
+                value.ifPresent(v -> entry.setField(field, v));
+            }
+
+            // Handle articleUrl and mrnumber fields separately, as they are non-nested properties in the JSON and can be retrieved as Strings directly
+            String doi = item.optString("articleUrl", "");
+            if (!doi.isEmpty()) {
+                entry.setField(StandardField.DOI, doi);
```

Comment:
Try to route through `DOI#parse`. If that optional is present, use `doi.getNormalized()`, otherwise use the `doi` variable. In That way, the http prefix is removed if it is a valid doi, but the full string is kept in case of a DOI parsing error.